dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/RateLimiterTest.java

```diff
@@ -0,0 +1,51 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import org.testng.annotations.Test;
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+import java.time.Duration;
+import java.lang.InterruptedException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.Callable;
+import java.util.Optional;
+
+public class RateLimiterTest {
```

Comment:
I would recommend revising this test.  Tests based upon synchronization using wall clock time tend to flap.  They also tend to make your test run longer.  For instance, this test alone will add at ;east another 23 seconds to every build.  Instead of using defined wall clock intervals to synchronize threads for testing its better to provide a deterministic mock clock to the threads and modify the clocks values during the test.  Bucket4j seems to have to ability to inject a custom clock, see https://bucket4j.com/8.1.1/toc.html#customizing-time-measurement-specify-custom-time-measurement-strategy .  It also implements mock clock in it own tests, https://github.com/bucket4j/bucket4j/blob/7477830cff62cef125226468814cf13fa51b530c/bucket4j-core/src/test/java/io/github/bucket4j/mock/TimeMeterMock.java#L7 .  I recommend we go the route of mocking the clock instead of adding sleeps for this test.